doc/tutorial/conv_arithmetic.txt

```diff
@@ -0,0 +1,834 @@
+convolutional layers (also known as fractionally strided convolutional layers)
+have been employed in more and more work as of late, and their relationship with
+convolutional layers has been explained with various degrees of clarity.
+
+The relationship between a convolution operation's input shape, kernel size,
+stride, padding and its output shape can be confusing at times.
+
+The tutorial's objective is twofold:
+
+* Explain the relationship between convolutional layers and transposed
+  convolutional layers.
+* Provide an intuitive understanding of the relationship between input shape,
+  kernel shape, zero padding, strides and output shape in convolutional and
+  transposed convolutional layers.
+
+Refresher: discrete convolutions
+================================
+
+The bread and butter of neural networks is *affine transformations*: a
+vector is received as input and is multiplied with a matrix to produce an
+output (to which a bias vector is usually added before passing the result
+through a nonlinearity). This is applicable to any type of input, be it an
+image, a sound clip or an unordered collection of features: whatever their
+dimensionality, their representation can always be flattened into a vector
+before the transformation.
+
+Images, sound clips and many other similar kinds of data have an intrinsic
+structure. More formally, they share these important properties:
+
+* They are stored as multi-dimensional arrays.
+* They feature one or more axes for which ordering matters (e.g., width and
+  height axes for an image, time axis for a sound clip).
+* One axis, called the channel axis, is used to access different views of the
+  data (e.g., the red, green and blue channels of a color image, or the left
+  and right channels of a stereo audio track).
+
+These properties are not exploited when an affine transformation is applied; in
+fact, all the axes are treated in the same way and the topological information
+is not taken into account. Still, taking advantage of the implicit structure of
+the data may prove very handy in solving some tasks, like computer vision and
+speech recognition, and in these cases it would be best to preserve it. This is
+where discrete convolutions come into play.
+
+A discrete convolution is a linear transformation that preserves this notion of
+ordering. It is sparse (only a few input units contribute to a given output
+unit) and reuses parameters (the same weights are applied to multiple locations
+in the input).
+
+Here is an example of a discrete convolution:
+
+.. figure:: conv_arithmetic_figures/numerical_no_padding_no_strides.gif
+
+The light blue grid is called the *input feature map*. To keep the drawing
+simple, a single input feature map is represented, but it is not uncommon to
+have multiple feature maps stacked one onto another. (An example of this is
+what was referred to earlier as *channels* for images and sound clips.)
+A *kernel* (shaded area) of value
```

Comment:
I think we should move " To keep the drawing simple, a single input feature map is represented, but it is not uncommon to have multiple feature maps stacked one onto another. (An example of this is what was referred to earlier as *channels* for images and sound clips.)" after [at the end of the paragraph](https://github.com/Theano/Theano/pull/4068/files#diff-119a4842baa8358cdc5cbc127b3bdaf9R151). 

It's better to introduce the terms *input feature map*, *kernel*, *output feature maps* altogether and then refer to the image. Furthermore, I would add that in the image a single kernel is applied and therefore a single output feature map is displayed. Finally, for completeness, I'd say that the output feature map is in green.